tests/stress/stress_test.go

```diff
@@ -358,16 +358,24 @@ func TestSync_Restart(t *testing.T) {
 }
 
 func TestPendingExtrinsic(t *testing.T) {
-	// TODO: Fix this test and enable it. Node syncing takes time.
-	t.Skip("skipping TestPendingExtrinsic")
-
 	t.Log("starting gossamer...")
 
 	utils.CreateConfigBabeMaxThreshold()
 
 	numNodes := 3
 	// index of node to submit tx to
-	idx := numNodes - 1 // TODO: randomize this
+	idx := numNodes - 1
+
+	// Randomize authority node selection.
+	rand.Seed(time.Now().UnixNano())
+	rand.Shuffle(len(utils.KeyList), func(i, j int) { utils.KeyList[i], utils.KeyList[j] = utils.KeyList[j], utils.KeyList[i] })
+
+	defer func() {
+		// Restore the key list.
+		sort.Slice(utils.KeyList, func(i, j int) bool {
+			return utils.KeyList[i] > utils.KeyList[j]
+		})
+	}()
```

Comment:
this doesn't actually submit it to a non-authority node, since it still submits it to `node` which is started with the BABE authority config

you can move starting the non-authority nodes up to when the authority node is started:
```
nodes, err := utils.InitializeAndStartNodes(t, numNodes-1, utils.GenesisDefault, utils.ConfigNoBABE)
nodes = append(nodes, node)
```

then start the `api` with a randomly chosen node index: 
```
api, err := gsrpc.NewSubstrateAPI(fmt.Sprintf("http://localhost:%s", nodes[randomIdx].RPCPort))
```

Reply:
@noot  I tried the above. Blocks are not getting propagated to other nodes. I was wondering if we can close this PR and I will investigate it in a separate issue?

Reply:
is the issue that the non-authority node can't validate the transaction? in that case, the original issue still applies so you can leave it open when this PR is closed. otherwise please open another issue